### ISSUE TITLE
Tests before submission

### ISSUE THREAD

**Issue by tfiedlerdev** [FranzSw/7-vs-capsule#3]
MNIST
- [x] run resnet2d
![image](https://user-images.githubusercontent.com/29177177/222451309-3a063b0d-7c75-4880-ad48-079ff588ed21.png)
Accuracy: 0.9788 after 10 epochs with command `python train.py resnet_2d --epochs=10 --slice_width=28 --learning_rate=0.001 --dataset=mnist`

- [x] run capsnet2d
works with `python train.py capsnet_2d --dataset=mnist --slice_width=128 --batch_size=2 --learning_rate=0.001` --> reaches accuracy 0.9693 after 1 epoch (however slice_width=28 does not work) 
![image](https://user-images.githubusercontent.com/29177177/222520204-ab1c4eba-e9e8-4fe8-a2bf-58a14590c93b.png)

  - [ ] make work for variable slice_width

LungPet2D
- [x] run capsnet2d
  - [x] Without undersampling`python train.py capsnet_2d --epochs=10 --slice_width=28 --learning_rate=0.01 --dataset=lungpetctx`
![image](https://user-images.githubusercontent.com/29177177/222470385-a537b3a4-e406-467b-8638-39b25cbe999a.png)
  - [x] With undersampling `python train.py capsnet_2d --epochs=10 --slice_width=128 --learning_rate=0.01 --dataset=lungpetctx --iterations=3 --class_imbalance=undersample --early_stopping`
![image](https://user-images.githubusercontent.com/29177177/222519909-3e7289c0-724f-4fb2-9146-cd7bc6728dda.png)
- [ ] run resnet2d 
  - [x] without undersampling `python train.py resnet_2d --epochs=10 --slice_width=28 --learning_rate=0.001 --dataset=lungpetctx --early_stopping`
![image](https://user-images.githubusercontent.com/29177177/222520041-df8a683a-f4e9-4855-8e10-c94ce7a37446.png)
  - [ ] with undersampling python train.py resnet_2d --epochs102 --slice_width=28 --learning_rate=0.001 --dataset=lungpetctx

LungPet3D
- [ ] ~~run resnet3d (to be implemented)~~
- [x] run capsnet3d
  - [x] Without undersampling `python train.py capsnet_3d --epochs=10 --slice_width=128 --learning_rate=0.01 --dataset=lungpetctx --iterations=3 --early_stopping --batch_size=4`
![image](https://user-images.githubusercontent.com/29177177/222894460-34f94a2d-1be3-4bde-94ca-b3a3bfce60ab.png)
  - [x] With undersampling `python train.py capsnet_3d --epochs=10 --slice_width=128 --learning_rate=0.01 --dataset=lungpetctx --iterations=3 --early_stopping --batch_size=4 --class_imbalance=undersample`
  
![image](https://user-images.githubusercontent.com/29177177/222903205-d4baff2d-84bb-44d4-93ed-f1406d1a9301.png)
